sdk/python/kfp/dsl/_python_component.py

```diff
@@ -47,3 +49,49 @@ def _python_component(func):
     return func
 
   return _python_component
+
+def component(func):
+  """Decorator for component functions that use ContainerOp.
+  This is useful to enable type checking in the DSL compiler
+
+  Usage:
+  ```python
+  @dsl.component
+  def foobar(model: TFModel(), step: MLStep()):
+    return dsl.ContainerOp()
+  """
+  def _component(*args, **kargs):
```

Comment:
Since your decorator does not have parameters, you can probably use a simpler way and return the same function. Se the comment below.